network/eni/eni.go

```diff
@@ -82,12 +82,7 @@ func (eni *ENI) AttachToLink() error {
 			return err
 		}
 
-		for _, i := range interfaces {
-			if vpc.CompareMACAddress(i.HardwareAddr, eni.macAddress) {
-				iface = &i
-				break
-			}
-		}
+		iface = getInterfaceByMac(eni.macAddress, interfaces)
```

Comment:
Should we move the net.Interfaces() call to the new getInterfaceByMACAddress function? That way we wouldn't have to pass it as an argument.

Reply:
that makes it hard to add unit test i think..